web/profiles/custom/os2loop/modules/os2loop_upvote/src/Helper/FlagHelper.php

```diff
@@ -3,25 +3,41 @@
 namespace Drupal\os2loop_upvote\Helper;
 
 use Drupal\Core\Cache\Cache;
+use Drupal\flag\FlagCountManager;
 
 /**
  * Flaghelper for creating top comment and count flags.
  */
 class FlagHelper {
+  /**
+   * The flag service.
+   *
+   * @var \Drupal\flag\FlagCountManager
+   */
+  protected $flagService;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\flag\FlagCountManager $flag_service
+   *   The flag service.
+   */
+  public function __construct(FlagCountManager $flag_service) {
+    $this->flagService = $flag_service;
+  }
 
   /**
    * Find the top comment.
    *
    * If there is a comment marked as "correct answer", this is it.
    * If there is no "correct answer", the top comment will be the most upvoted.
    */
-  public function preprocessFlag(array &$variables) {
-    $flag_service = \Drupal::service('flag.count');
+  public function preprocessField(array &$variables) {
```

Comment:
Consider adding `Implements hook_preprocess_field()` at the top of the comment to make it clear that this function actually does that.